lib/chamber/commands/securable.rb

```diff
@@ -52,11 +52,10 @@ def ignored_settings_filepaths
       Shellwords.escape(filename)
     end
 
-    `
-      git ls-files --other --ignored --exclude-per-directory=.gitignore |
-      sed -e "s|^|#{Shellwords.escape(rootpath.to_s)}/|" |
-      grep --colour=never -E '#{shell_escaped_chamber_filenames.join('|')}'
-    `.split("\n")
+    `git ls-files --other --ignored --exclude-per-directory=.gitignore`
```

Comment:
It'd ideally remove the `git` dependency, too.

If I understand correctly from the `#ignored_settings_filepaths` method name, the code is supposed to return absolute paths to all existing _settings*.yml_ that are present in the `.gitignore`?

If that's the case, it could also be done via plain Ruby. Here's a pseudo-code that'd replace my current proposed changes of the idea approach I have in mind:
```
Dir["#{rootpath}/**/*/.gitignore"].filter_map do |gitignore|
  lines = File.read(gitignore).split("\n")
  shell_escaped_chamber_filenames.select { |filename| lines.include?(filename) }
end.flatten.uniq
```
Sorry if I misunderstood anything, please advise.

Reply:
**Note:** If we go this route, there's also `git config --get 'user.name'` in _lib/chamber/binary/runner.rb_

Reply:
I'm not really interested in getting rid of `git`.  The entire point of this library is so that you're not checking in your secrets into version control.  The only thing that would convince me is if there was a version compatibility issue or someone wanted to add support for something other than `git`.

Reply:
I don't believe that's **the entire point** of this library. Tbh I don't think that's even the main point of this library - that'd be, in my understanding, the environment variables management. One of the library's features is safe keeping secrets in the version control.

I encourage you to consider the least dependency approach, at the end of the day its just less headache. Especially that there's someone who wishes to do that for you, under little guidance. Your call.

Reply:
I would guess that if you look at the implementation of `git ls-files` it is going to be far more complicated than a couple lines of code.  It's easy to look at something and say "Oh this is just x" but the trouble comes in all the edge cases you're not thinking about.  At which point it becomes something that has to be maintained.  Right now, `git` maintains that code.  I do not want to take on that burden.  I'm happy for the help but at the end of the day, I'm the one that's going to end up being responsible for maintaining that code and I have little time as it is.

Reply:
Your removal of `sed` and `grep` looks good though.  I'll get that into 3.0.

Reply:
Sure, your call.